### PR TITLE
Enable Python pre-commit Hooks

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This PR enables almost all of the Python related [pre-commit](https://pre-commit.com) hooks in our configuration. Any changes required to pass those hooks are also made.

## 💭 Motivation and context ##

It's good for projects to conform to our standard linting suite whenever possible. The monolithic and older nature of this repository has been a deterrent to enabling the full suite of hooks we have in our configuration.

N.B. The `mypy` hook remains disabled because it is designed to run against Python packages. Enabling results in the following error:

```console
$ pre-commit run mypy --all-files
mypy.....................................................................Failed
- hook id: mypy
- exit code: 2

terraform_nessus_only/configure.py: error: Duplicate module named "configure" (also at "terraform/configure.py")
terraform_nessus_only/configure.py: note: Are you missing an __init__.py? Alternatively, consider using --exclude to avoid checking one of them.
Found 1 error in 1 file (errors prevented further checking)
```

If we want to exclude one or both of the `configure.py` files that would work. Alternatively we can rename them so they are distinct. I would be happy to see opinions so we can get that hook enabled in this PR.

Edit: I have found a workaround that isn't outlandish.

## 🧪 Testing ##

Automated tests pass. 🎉

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]

**Comment by jsf9k** [comment]
Great start on breaking this horse!

**Review by jsf9k — APPROVED** [review]
Nice workaround!

File: terraform_egress_pub/scripts/egress_pub.py [hunk 1]
```diff
@@ -1,13 +1,14 @@
 #!/usr/bin/env python3
-import boto3
+
+"""Gather and publich the list of IPs of EC2 instances tagged for publication."""
```
Comment (dav3r) [inline terraform_egress_pub/scripts/egress_pub.py]: Typo:
```suggestion
"""Gather and publish the list of IPs of EC2 instances tagged for publication."""
```

**Review by dav3r — APPROVED** [review]
Very nice - good workaround! ⚡

**Pull request merged**